components/modal/src/Modal.tsx

```diff
@@ -182,84 +172,81 @@ export const Modal = ({
 		}
 	}, [onClose]);
 
-	// on escape close modal
-	const keyHandler = useCallback(
-		(e: KeyboardEvent) => {
-			if (dismissible && open && e.keyCode === 27) handleClose();
-		},
-		[dismissible, handleClose, open]
-	);
-
-	// bind key events
 	useEffect(() => {
-		window.document.addEventListener('keydown', keyHandler);
-		return () => {
-			window.document.removeEventListener('keydown', keyHandler);
-		};
-		// eslint-disable-next-line react-hooks/exhaustive-deps
+		setMounted(true);
 	}, []);
 
-	if (typeof window !== 'undefined') {
-		return ReactDOM.createPortal(
-			<GEL brand={brand}>
-				<ModalContext.Provider value={{ state }}>
-					<FocusOn enabled={open}>
-						<Modal
-							ref={modalRef}
-							onClick={(e: MouseEvent) => {
-								if (e.target !== e.currentTarget) return;
-								if (dismissible) {
-									handleClose();
-								}
-							}}
-							state={state}
-							{...rest}
-							{...modalAttributes(state)}
-							css={modalStyles(state)}
-						>
-							<ModalDialog
-								state={state}
-								{...modalDialogAttributes(state)}
-								css={modalDialogStyles(state)}
+	// These styles call hooks and will create hook call warning if called directly in component
+	const modalDialogStyle = modalDialogStyles(state);
+	const headerStyle = headerStyles(state);
+	const headingStyle = headingStyles(state);
+	const closeBtnStyle = closeBtnStyles(state);
```

Comment:
These need to be called outside of the component as they contain the useBrand hook, if they are called like some of the others inside the component it will create warnings about unexpected hook calls

Reply:
This should be the other way around. A hook must be called within a component otherwise React will complain about unexpected calls.

Reply:
I may have worded that the wrong way, if they are called inside the return it was causing unexpected calls warnings. They are still being called in the component, here is an example of what was happening:
<img width="165" alt="image" src="https://github.com/WestpacGEL/GEL/assets/46306101/d56220a7-b0af-4f44-adfc-306949c1bb47">
<img width="422" alt="image" src="https://github.com/WestpacGEL/GEL/assets/46306101/51d5d469-4153-43ad-be56-ab916d907fe8">